insights/parsers/xfs_db.py

```diff
@@ -54,3 +59,58 @@ def parse_content(self, content):
 
         if not self.fragmentation_factor:
             raise SkipComponent("Invalid output of xfs fragmentation")
+
+
+@parser(Specs.xfs_db_freesp)
+class XFSDbFreesp(CommandParser):
+    """
+    The ``/usr/sbin/xfs_db -r -c freesp <mounted_device>`` command provides free space information
+    for the xfs filesystem.
+
+    Sample output::
+
+        from      to extents  blocks    pct
+           1       1      16      16   0.01
+           2       3       1       2   0.00
+          64     127       1     103   0.04
+        32768   65536      4   23113   99.95
+
+    Attributes:
+        free_stat(list of dict): A list of dictionaries with the key-value data from the table.
+
+    Examples:
+        >>> type(xfs_db_freesp)
+        <class 'insights.parsers.xfs_db.XFSDbFreesp'>
+        >>> len(xfs_db_freesp.free_stat) == 4
+        True
+        >>> xfs_db_freesp.free_stat[0]['from']
+        '1'
+        >>> len(xfs_db_freesp.search(pct='0.00')) == 1
+        True
+    """
+
+    def parse_content(self, content):
+        keys = ["from", "to", "extents", "blocks", "pct"]
+        header_line = content[0].split()
+
+        if header_line != keys:
```

Comment:
It's not so `robust` to assert the first line to be the header_line, in fact the `parse_delimited_table` provides options to skip those non-header-lines, e.g. `parse_delimited_table(content,heading_ignore('from')`

Reply:
@xiangce Created a new commit to enhance the robust, please help take a look again. Thx!